pkg/operator/watchdog/cmd.go

```diff
@@ -0,0 +1,247 @@
+package watchdog
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/config/client"
+	"github.com/openshift/library-go/pkg/controller/fileobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+)
+
+type FileWatcherOptions struct {
+	ProcessName string
+	Files       []string
+	KubeConfig  string
+	Namespace   string
+	Interval    time.Duration
+
+	recorder       events.Recorder
+	initialContent map[string][]byte
+
+	// for unit-test
+	getPIDByNameFn       func(name string) (int, bool, error)
+	signalFn             func(pid int) error
+	setFilesPathPrefixFn func([]string, int) []string
+}
+
+func NewFileWatcherOptions() *FileWatcherOptions {
+	return &FileWatcherOptions{
+		getPIDByNameFn:       GetPidByName,
+		setFilesPathPrefixFn: setFilesPathPrefix,
+		signalFn: func(pid int) error {
+			return syscall.Kill(pid, syscall.SIGTERM)
+		},
+	}
+}
+
+// NewFileWatcherWatchdog return the file watcher watchdog command.
+func NewFileWatcherWatchdog() *cobra.Command {
+	o := NewFileWatcherOptions()
+
+	cmd := &cobra.Command{
+		Use:   "file-watcher-watchdog",
+		Short: "Watch files on the disk and kill the specified process on change",
+		Run: func(cmd *cobra.Command, args []string) {
+			klog.V(1).Info(cmd.Flags())
+			klog.V(1).Info(spew.Sdump(o))
+
+			ctx := context.TODO()
+
+			if err := o.Complete(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := o.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+
+			if err := o.Run(ctx); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	o.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *FileWatcherOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.ProcessName, "process-name", "", "name of the process to send signal to on file change (eg. 'hyperkube').")
+	fs.StringSliceVar(&o.Files, "files", o.Files, "comma separated list of file names to monitor for changes")
+	fs.StringVar(&o.KubeConfig, "kubeconfig", o.KubeConfig, "kubeconfig file or empty")
+	fs.StringVar(&o.Namespace, "namespace", o.Namespace, "namespace to report the watchdog events")
```

Comment:
this is used only for events... as this will restart on every change, I think we **need** events to confirm this works as expected.